### PR TITLE
BREAKING: support exclude `kinds/namespaces` and include `kinds/namespaces`

### LINKED ISSUE #6195: trivy k8s: new flags `--include-namespaces` and `--exclude-namespaces`
it is required to add two new flags `--include-namespaces` and `--exclude-namespaces`

when `--include-namespaces` is specified, example `--include-namespaces test` meaning the scan result will resent finding only from `test` namespace

when `--exclude-namespaces` is specified, example `--exclude-namespaces test` meaning the scan result will resent finding from all namespaces except of `test` namespace

both flags `--include-namespaces` and `--exclude-namespaces` are not allowed to be specified in conjunction on same scan

### LINKED ISSUE #6196: trivy k8s: new flags `--include-kinds` and `--exclude-kinds`
it is required to add two new flags `--include-kinds` and `--exclude-kinds`

when `--include-kinds` is specified, example `--include-kinds node` meaning that the scan result will resent finding only from `node` kind

when `--exclude-kinds` is specified, example `--exclude-kinds node` meaning the scan result will resent finding from all namespaces except of `node` kind

both flags `--include-kinds` and `--exclude-kinds` are not allowed to be specified in conjunction on same scan

### LINKED ISSUE #6467: trivy k8s: remove `cluster` and `all` args and unused flags
it is required as part of new `trivy k8s...` experience  to:

## Remove the support of the following args:
- `cluster` 
-  `all` 
## Remove the unused flags:
- context
- namespace
- all-namespaces

### PR DESCRIPTION
**Pull request by chen-keinan**
## Description
support exclude kinds/namespaces and include kinds/namespaces

## Related issues
- Close #6195 
- Close #6196
- Close #6467 
- Related https://github.com/aquasecurity/trivy-kubernetes/pull/311

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.

### CODE REVIEW

**Review by simar7 — APPROVED** [review]

File: docs/docs/references/configuration/cli/trivy_kubernetes.md [hunk 1]
```diff
@@ -36,11 +37,12 @@ trivy kubernetes [flags] { cluster | all | specific resources like kubectl. eg:
       --components strings                specify which components to scan (workload,infra) (default [workload,infra])
       --config-data strings               specify paths from which data for the Rego policies will be recursively loaded
       --config-policy strings             specify the paths to the Rego policy files or to the directories containing them, applying config files
-      --context string                    specify a context to scan
```
Comment (knqyf263) [inline docs/docs/references/configuration/cli/trivy_kubernetes.md:39]: I'm sorry I was on holiday so I haven't followed the discussion, but what is the motivation of removing the `--context` flag and making it a CLI argument?

**Review by knqyf263 — COMMENTED** [review]
We also need to update the document, right?
https://aquasecurity.github.io/trivy/v0.50/docs/target/kubernetes/

File: docs/docs/references/configuration/cli/trivy_kubernetes.md (hunk 1 above)
Comment (chen-keinan) [reply docs/docs/references/configuration/cli/trivy_kubernetes.md:39]: @knqyf263 we decided to do this in order to keep the convention with other trivy command which is:

trivy <target> <target name>

examples:
trivy image <image name>
trivy k8s <context>

Comment (chen-keinan) [reply docs/docs/references/configuration/cli/trivy_kubernetes.md:39]: if context not specified we will use the default

File: pkg/k8s/commands/run.go [hunk 2]
```diff
@@ -20,48 +20,30 @@ import (
...
+	}
+	if len(args) > 0 {
+		clusterOptions = append(clusterOptions, k8s.WithContext(args[0]))
+	}
+	cluster, err := k8s.GetCluster(clusterOptions...)
 	if err != nil {
 		return xerrors.Errorf("failed getting k8s cluster: %w", err)
 	}
+	ctx, cancel := context.WithTimeout(ctx, opts.Timeout)
+	defer cancel()
```
Comment (simar7) [inline pkg/k8s/commands/run.go:38]: nit: you can use the existing defer func() to call this `cancel()`.

**Comment by simar7** [comment]
@chen-keinan reviewed the latest changes, left one comment.

**Comment by knqyf263** [comment]
@chen-keinan Are you planning to update the doc?
https://github.com/aquasecurity/trivy/pull/6323#pullrequestreview-2000637878

**Comment by chen-keinan** [comment]
> @chen-keinan Are you planning to update the doc? [#6323 (review)](https://github.com/aquasecurity/trivy/pull/6323#pullrequestreview-2000637878)

there is a specific PR #6465 for docs

**Comment by knqyf263** [comment]
I think it's better to include the document update regarding including/excluding in this PR. It makes reverting/blaming easier. What do you think? It may be okay this time, but IMHO, it is generally better. It is common to forget to update the documentation and unintentionally end up with a different PR 😄

**Comment by chen-keinan** [comment]
> I think it's better to include the document update regarding including/excluding in this PR. It makes reverting/blaming easier. What do you think? It may be okay this time, but IMHO, it is generally better. It is common to forget to update the documentation and unintentionally end up with a different PR 😄

@knqyf263 the docs PR is for all experience changes not just for this one , that why we make it in a separate PR

**Comment by knqyf263** [comment]
Anyway, it's already approved by Simar. You can merge whenever.

**Pull request merged**